mongo/integration/collection_test.go

```diff
@@ -1861,6 +1863,334 @@ func TestCollection(t *testing.T) {
+		return bson.RawValue{
+			Type:  valType,
+			Value: data,
+		}
+	}
+
+	boolPtr := func(b bool) *bool {
+		return &b
+	}
+
+	mt.Run("insert one", func(mt *mtest.T) {
+		doc := bson.D{{"x", 42}}
+
+		testCases := []struct {
+			name     string
+			opts     *options.InsertOneOptions
+			expected bson.RawValue
+		}{
+			{
+				"empty",
+				nil,
+				bson.RawValue{},
+			},
+			{
+				"false",
+				&options.InsertOneOptions{BypassEmptyTsReplacement: boolPtr(false)},
+				marshalValue(false),
+			},
+			{
+				"true",
+				&options.InsertOneOptions{BypassEmptyTsReplacement: boolPtr(true)},
+				marshalValue(true),
+			},
+		}
+		for _, tc := range testCases {
+			mt.Run(tc.name, func(mt *mtest.T) {
+				_, err := mt.Coll.InsertOne(context.Background(), doc, tc.opts)
+				require.NoError(mt, err, "InsertOne error: %v", err)
+				evt := mt.GetStartedEvent()
+				val := evt.Command.Lookup("bypassEmptyTsReplacement")
+				require.Equal(mt, tc.expected, val, "expected bypassEmptyTsReplacement to be %s", tc.expected.String())
+			})
+		}
+	})
+	mt.Run("insert many", func(mt *mtest.T) {
+		docs := []interface{}{
+			bson.D{{"x", 42}},
+			bson.D{{"y", "foo"}},
+		}
+
+		testCases := []struct {
+			name     string
+			opts     *options.InsertManyOptions
+			expected bson.RawValue
+		}{
+			{
+				"empty",
+				nil,
+				bson.RawValue{},
+			},
+			{
+				"false",
+				&options.InsertManyOptions{BypassEmptyTsReplacement: boolPtr(false)},
+				marshalValue(false),
+			},
+			{
+				"true",
+				&options.InsertManyOptions{BypassEmptyTsReplacement: boolPtr(true)},
+				marshalValue(true),
+			},
+		}
+		for _, tc := range testCases {
+			mt.Run(tc.name, func(mt *mtest.T) {
+				_, err := mt.Coll.InsertMany(context.Background(), docs, tc.opts)
+				require.NoError(mt, err, "InsertMany error: %v", err)
+				evt := mt.GetStartedEvent()
+				val := evt.Command.Lookup("bypassEmptyTsReplacement")
+				require.Equal(mt, tc.expected, val, "expected bypassEmptyTsReplacement to be %s", tc.expected.String())
+			})
+		}
+	})
+	mt.Run("update one", func(mt *mtest.T) {
+		filter := bson.D{{"x", 42}}
+		update := bson.D{{"$inc", bson.D{{"x", 1}}}}
+
+		testCases := []struct {
+			name     string
+			opts     *options.UpdateOptions
+			expected bson.RawValue
+		}{
+			{
+				"empty",
+				nil,
+				bson.RawValue{},
+			},
+			{
+				"false",
+				&options.UpdateOptions{BypassEmptyTsReplacement: boolPtr(false)},
+				marshalValue(false),
+			},
+			{
+				"true",
+				&options.UpdateOptions{BypassEmptyTsReplacement: boolPtr(true)},
+				marshalValue(true),
+			},
+		}
+		for _, tc := range testCases {
+			mt.Run(tc.name, func(mt *mtest.T) {
+				_, err := mt.Coll.UpdateOne(context.Background(), filter, update, tc.opts)
+				require.NoError(mt, err, "UpdateOne error: %v", err)
+				evt := mt.GetStartedEvent()
+				val := evt.Command.Lookup("bypassEmptyTsReplacement")
+				require.Equal(mt, tc.expected, val, "expected bypassEmptyTsReplacement to be %s", tc.expected.String())
```

Comment:
```suggestion
				assert.Equal(mt, tc.expected, val, "expected bypassEmptyTsReplacement to be %s", tc.expected.String())
```
Suggest not stoping the test if we get an unexpected result we are testing for.